webpack/scenes/ContentViews/Table/ContentViewsTable.js

```diff
@@ -48,47 +42,64 @@ const ContentViewTable = ({
 
   const onExpand = (_event, rowIndex, colIndex, isOpen) => {
     const { rows } = table;
-    const contentViewId = cvIdFromRow(rowIndex);
     // adjust for the selection checkbox cell being counted in the index
     const adjustedColIndex = colIndex - 1;
 
     if (!isOpen) {
       setRowMapping((prev) => {
-        const updatedMap = { ...prev[contentViewId], expandedColumn: adjustedColIndex };
-        return { ...prev, [contentViewId]: updatedMap };
+        const updatedMap = { ...prev[rowIndex], expandedColumn: adjustedColIndex };
+        return { ...prev, [rowIndex]: updatedMap };
       });
     } else {
-      // remove the row completely by assigning it to a throwaway variable
-      // eslint-disable-next-line camelcase, no-unused-vars
-      const { [contentViewId]: _throwaway, ...newMap } = rowMapping;
-      setRowMapping(newMap);
+      setRowMapping(prev => ({ ...prev, [rowIndex]: {} }));
     }
 
     setTable(prevTable => ({ ...prevTable, rows }));
   };
 
+  // Prevents flash of "No Content" before rows are loaded
+  const tableStatus = () => {
+    if (typeof results === 'undefined') return status; // will handle errored state
+    const resultsLength = results && results.length;
+    const rowMappingLength = Object.keys(rowMapping) && Object.keys(rowMapping).length;
+    if (resultsLength === 0) return status; // handles no CVs present
+    if (rowMappingLength > 0 && resultsLength > 0) {
+      const rowMappingIds = Object.values(rowMapping).map(map => map.id);
+      const resultsIds = Array.from(results.map(result => result.id));
```

Comment:
If you are wondering why I need `Array.from`, the results object is an `Immutable` object